src/util/taproot.rs

```diff
@@ -558,14 +558,13 @@ impl TaprootMerkleBranch {
                 sl.len() / TAPROOT_CONTROL_NODE_SIZE,
             ))
         } else {
-            let mut inner = Vec::with_capacity(sl.len() / TAPROOT_CONTROL_NODE_SIZE);
-            let mut start = 0;
-            while start < sl.len() {
-                let elem = sha256::Hash::from_slice(&sl[start..start + TAPROOT_CONTROL_NODE_SIZE])
-                    .expect("checked above for size 32");
-                inner.push(elem);
-                start += TAPROOT_CONTROL_NODE_SIZE;
-            }
+            let inner = sl
+                .chunks(TAPROOT_CONTROL_NODE_SIZE)
```

Comment:
`chunks_exact` is faster though since we know the size...

Reply:
But it is rust 1.31 :(, outside our current MSRV of 1.29

Reply:
Added to the list in #510